File: frontend/src/types/callAnalytics.ts
export interface CallAnalyticsRecord {
  id: string
  time_from: string
  time_to: string
  extension_number: string
  extension_name: string
  total_calls: number
  inbound: number
  outbound: number
  direct: number
  from_queue: number
  transferred: number
  portal_equiv: number
  duration_seconds: number
  external: number
  internal: number
  answered: number
  not_answered: number
  completed: number
  abandoned: number
  voicemail: number
  created_at: string
  updated_at: string
  deleted_at: string | null
}

export interface CallAnalyticsKpis {
  total_calls: number
  total_inbound: number
  total_outbound: number
  total_answered: number
  total_not_answered: number
  total_completed: number
  total_abandoned: number
  total_voicemail: number
  total_duration_seconds: number
  extension_count: number
}

export interface CallByExtension {
  extension_number: string
  extension_name: string
  total_calls: number
  inbound: number
  outbound: number
  answered: number
  not_answered: number
  completed: number
  abandoned: number
  voicemail: number
  duration_seconds: number
}

export interface FetchResult {
  rows_saved: number
  time_from: string
  time_to: string
}

export interface IsoRange {
  from: string
  to: string
}

File: backend/app/schemas/call_analytics.py
from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class CallAnalyticsBase(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    time_from:        datetime
    time_to:          datetime
    extension_number: str
    extension_name:   str
    total_calls:      int
    inbound:          int
    outbound:         int
    direct:           int
    from_queue:       int
    transferred:      int
    portal_equiv:     int
    duration_seconds: int
    external:         int
    internal:         int
    answered:         int
    not_answered:     int
    completed:        int
    abandoned:        int
    voicemail:        int


class CallAnalyticsOut(CallAnalyticsBase):
    id:         uuid.UUID
    created_at: datetime
    updated_at: datetime
    deleted_at: datetime | None


class CallAnalyticsKpisOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    total_calls:            int
    total_inbound:          int
    total_outbound:         int
    total_answered:         int
    total_not_answered:     int
    total_completed:        int
    total_abandoned:        int
    total_voicemail:        int
    total_duration_seconds: int
    extension_count:        int


class CallAnalyticsByExtensionOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    extension_number: str
    extension_name:   str
    total_calls:      int
    inbound:          int
    outbound:         int
    answered:         int
    not_answered:     int
    completed:        int
    abandoned:        int
    voicemail:        int
    duration_seconds: int


class FetchRequest(BaseModel):
    time_from: datetime
    time_to:   datetime


class FetchResultOut(BaseModel):
    rows_saved: int
    time_from:  datetime
    time_to:    datetime
